### ISSUE TITLE
Sample of Impersonating User

### ISSUE THREAD

**Issue by rwinch** [spring-projects/spring-session#170]
Demonstrate how one can use the multiple session support to impersonate a user to see what a particular type of user experiences. For example, admin is troubleshooting an issue, previewing permissions, etc.

**Comment by chrylis** [comment]
So basically Spring Sudo?

**Comment by rwinch** [comment]
Not really. It is pretty much the same idea as the users sample but instead of authenticating with multiple users (via some external mechanism) impersonation is done. The reason it is different is that the sudo is external of Spring Session (it does not care about how the user is authenticated).

**Comment by ptahchiev** [comment]
Any updates on this one - I'd really love to see this documented to give it a try :) Is it possible to this now? Is it just the documentation that is missing, or does it need extra development?

**Comment by rwinch** [comment]
Look at the multi session example, but instead of requiring authentication just create a user and set it in the session

**Comment by ptahchiev** [comment]
Hey @rwinch  I can't seem to find the multi session example? Also the multi-session functionality seems to be missing. Was it dropped from the code base? Previously it was in the `CookieHttpSessionStrategy` but now it isn't any more. Regarding your comment Creating a `UserDetails` object is easy, but what does `set it in the session` mean? Do i set the user as an attribute of the session? Under what name?

**Comment by vpavic** [comment]
@ptahchiev See #906 and #917.
